app/transport/http/middleware.go

```diff
@@ -0,0 +1,33 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/go-devs-ua/octagon/lgr"
+)
+
+// Middleware is simple decorator
+type Middleware func(http.Handler) http.Handler
+
+// WrapMiddleware will build middleware chain
+func WrapMiddleware(handler http.Handler, middleware ...Middleware) http.Handler {
+	for _, m := range middleware {
+		handler = m(handler)
+	}
+	return handler
+}
+
+// WithLogRequest will log detailed request info
+func WithLogRequest(logger *lgr.Logger) Middleware {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			defer logger.Infow("Request:",
+				"Header", req.Header,
```

Comment:
So, what do we expect to receive in "Header"?

Reply:
I mean, we print all headers now. Why?

Reply:
Moreover, "User-Agent" is also comes from Header map, so its value is duplicated now.

Reply:
Please recall the principal that we discussed on the previous call - minimalism.
You want to print only data that can be useful. 
Cos right now, if user add 100 custom headers - you will print them 😄

Reply:
Amateur talent activities were removed successfully :smiley: